src/utils/update/trust.model.fetch.ts

```diff
@@ -0,0 +1,334 @@
+    notified_date_month: appointmentDate?.month ?? "",
+    notified_date_year: appointmentDate?.year ?? "",
+    corporate_indicator: yesNoResponse.Yes
+  };
+  trust.HISTORICAL_BO?.push(historicalCorporateTrustee);
+};
+
+export const retrieveTrustLinks = async (req: Request, appData: ApplicationData) => {
+  const overseasEntityId = appData.overseas_entity_id;
+  const transactionId = appData.transaction_id;
+
+  if (transactionId === undefined || overseasEntityId === undefined) {
+    logger.errorRequest(req, "Trust links could not be retrieved as transactionId or overseasEntityId is undefined");
+    return;
+  }
+
+  const trustLinks = await getTrustLinks(req, transactionId, overseasEntityId);
+
+  if (trustLinks === undefined || trustLinks.length === 0) {
+    logger.info(`No trust links found for overseas entity ${overseasEntityId}`);
+    return;
+  }
+
+  for (const trustLink of trustLinks) {
+    logger.debug("Loaded trust link " + trustLink.hashedTrustId + " for coprporate appointment " + trustLink.hashedCorporateBodyAppointmentId);
+    mapTrustLink(trustLink, appData);
+  }
+};
+
+export const mapTrustLink = (trustLink: TrustLinkData, appData: ApplicationData) => {
+  const trust = appData.update?.review_trusts?.find(t => t.ch_reference === trustLink.hashedTrustId);
+  if (trust) {
+    const individualBeneficialOwner = appData.beneficial_owners_individual?.find((beneficialOwner) => beneficialOwner.ch_reference === trustLink.hashedCorporateBodyAppointmentId);
+    if (individualBeneficialOwner) {
+      logger.debug("Linking individual beneficial owner " + individualBeneficialOwner.ch_reference + " to trust " + trust.ch_reference);
+      if (individualBeneficialOwner.trust_ids === undefined) {
+        individualBeneficialOwner.trust_ids = [];
+      }
+      individualBeneficialOwner.trust_ids.push(trust.trust_id);
+    } else {
+      const corporateBeneficialOwner = appData.beneficial_owners_corporate?.find((beneficialOwner) => beneficialOwner.ch_reference === trustLink.hashedCorporateBodyAppointmentId);
+      if (corporateBeneficialOwner) {
+        logger.debug("Linking corporate beneficial owner " + corporateBeneficialOwner.ch_reference + " to trust " + trust.ch_reference);
+        if (corporateBeneficialOwner.trust_ids === undefined) {
+          corporateBeneficialOwner.trust_ids = [];
+        }
+        corporateBeneficialOwner.trust_ids.push(trust.trust_id);
+      }
+    }
+  }
+};
+
+const mapTrusteeType = (trusteeTypeId: string): RoleWithinTrustType => {
+  switch (trusteeTypeId) {
+      case "5005":
+        return RoleWithinTrustType.INTERESTED_PERSON;
+      case "5004":
+        return RoleWithinTrustType.GRANTOR;
+      case "5003":
+        return RoleWithinTrustType.SETTLOR;
+      case "5002":
+      default:
+        return RoleWithinTrustType.BENEFICIARY;
+  }
+};
+
+const mapResidentialAddress = (trustee: TrustCorporate, trusteeData: CorporateTrusteeData) => {
```

Comment:
Change this to mapRegisteredOfficeAddress

Reply:
good spot. I'm blaming copilot from now on.